kolibri/plugins/management/assets/src/device_management/state/actions/availableChannelsActions.js

```diff
@@ -48,3 +52,25 @@ export function showAvailableChannelsPage(store) {
 export function getRemoteChannelByToken(token) {
   return RemoteChannelResource.getModel(token).fetch()._promise;
 }
+
+/**
+ * Makes a request to Kolibri Studio to get info on unlisted channels. And append
+ * them to the public channels.
+ *
+ * @param {Array<Channel>} publicChannels - the list of publich channels, which will not be queried
+ * @returns {Promise<Array<Channel>>}
+ */
+export function getAllRemoteChannels(store, publicChannels) {
+  const installedChannels = installedChannelList(store.state);
+  const potentiallyUnlisted = differenceBy(installedChannels, publicChannels, 'id').filter(
```

Comment:
A little bit confused why this is necessary, when the data we get back from content curation should have a `public` flag on it to indicate that the channel is public, is this not being passed through from Kolibri to the frontend? https://github.com/fle-internal/content-curation/blob/develop/contentcuration/contentcuration/serializers.py#L708

Reply:
Kolibri Studio's Channel's API only returns public channels (i.e. where `public: true`). The only API available to check on unlisted channels (`public: false`) is the lookup endpoint.

Don't know if this is possible, but if there were a way query the Channel's API to get both the public and unlisted-but-intalled-on-my-device channels, that would be nice.

Reply:
Seems like this data is being sent up to the front end also: https://github.com/learningequality/kolibri/blob/release-v0.7.x/kolibri/content/api.py#L401

Reply:
Yeah, that's how the publicChannels are represented. But there is no obvious way to get the very specific group of unlisted-but-installed-on-my-device channels. If there were, then we could get all such channels in one go, and I would just read off the `public` flag.

Reply:
The main issue is that the ContentMetadata does not have this `public` attribute.

Reply:
Ah, ok - got it now. OK, let's leave this approach as is then, until we can make the API work more neatly for the frontend.